contrib/sbtc/token-key-admin/contracts/sbtc-token-key-admin.clar

```diff
@@ -0,0 +1,186 @@
+
+;; title: sbtc-token-key-admin
+;; version:
+;; summary:
+;; description:
+
+;; traits
+;;
+(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
+
+;; token definitions
+;; 
+(define-fungible-token sbtc u21000000)
+
+;; constants
+;;
+(define-constant err-invalid-caller u1)
+(define-constant err-invalid-signer-id u2)
+(define-constant err-not-token-owner u3)
+
+;; data vars
+;;
+(define-data-var contract-owner principal tx-sender)
+(define-data-var coordinator (optional {addr: principal, key: (buff 33)}) none)
+(define-data-var num-keys uint u4000)
+(define-data-var num-parties uint u4000)
+(define-data-var threshold uint u2800)
+(define-data-var bitcoin-payout-address (optional (string-ascii 72)) none)
+
+;; data maps
+;;
+(define-map signers uint {addr: principal, key: (buff 33)})
+
+;; public functions
+;;
+(define-public (set-contract-owner (owner principal))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set contract-owner owner))
+    )
+)
+
+(define-public (set-coordinator-data (data {addr: principal, key: (buff 33)}))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set coordinator (some data)))
+    )
+)
+
+(define-public (set-num-keys (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set num-keys n))
+    )
+)
+
+(define-public (set-num-parties (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set num-parties n))
+    )
+)
+
+(define-public (set-threshold (n uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set threshold n))
+    )
+)
+
+(define-public (set-signer-data (id uint) (data {addr: principal, key: (buff 33)}))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (asserts! (is-valid-signer-id id) (err err-invalid-signer-id))
+        (ok (map-set signers id data))
+    )
+)
+
+(define-public (delete-signer-data (id uint))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (asserts! (is-valid-signer-id id) (err err-invalid-signer-id))
+        (ok (map-delete signers id))
+    )
+)
+
+(define-public (set-bitcoin-payout-address (addr (string-ascii 72)))
+    (begin
+        (asserts! (is-contract-owner) (err err-invalid-caller))
+        (ok (var-set bitcoin-payout-address (some addr)))
+    )
+)
+
+(define-public (mint! (amount uint))
```

Comment:
I'm thinking we should add a string memo argument to mint! to record the associated Bitcoin peg-in request txid.